temporal/server.go

```diff
@@ -386,44 +383,39 @@ func (s *Server) newBootstrapParams(
 	return params, nil
 }
 
-func (s *Server) getESConfigClient(advancedVisibilityWritingMode string) (*config.Elasticsearch, esclient.Client, error) {
-	if advancedVisibilityWritingMode == common.AdvancedVisibilityWritingModeOff {
+func (s *Server) getESConfigAndClient() (*esclient.Config, esclient.Client, error) {
+	if !s.so.config.Persistence.AdvancedVisibilityConfigExist() {
 		return nil, nil, nil
 	}
 
 	advancedVisibilityStore, ok := s.so.config.Persistence.DataStores[s.so.config.Persistence.AdvancedVisibilityStore]
 	if !ok {
-		return nil, nil, fmt.Errorf("unable to find advanced visibility store in config for %q key", s.so.config.Persistence.AdvancedVisibilityStore)
-	}
-
-	indexName := advancedVisibilityStore.ElasticSearch.GetVisibilityIndex()
-	if len(indexName) == 0 {
-		return nil, nil, errors.New("visibility index in missing in Elasticsearch config")
+		return nil, nil, nil
```

Comment:
shouldn't you return error here?

Reply:
At this moment, config is already validated and this should never happened. If `AdvancedVisibilityConfigExist` (checked few lines above) it is valid and must present. But I agree to return error here.